src/Microsoft.CodeDom.Providers.DotNetCompilerPlatform/Util/CompilationSettings.cs

```diff
@@ -56,8 +59,17 @@ internal static class CompilationSettingsHelper {
                 ttl = DefaultCompilerServerTTLInDevEnvironment;
             }
 
-            _csc = new CompilerSettings(CompilerFullPath(@"bin\roslyn\csc.exe"), ttl);
-            _vb = new CompilerSettings(CompilerFullPath(@"bin\roslyn\vbc.exe"), ttl);
+            var customRoslynCompilerLocation = Environment.GetEnvironmentVariable(CustomRoslynCompilerLocation, EnvironmentVariableTarget.Process);
+            if(customRoslynCompilerLocation != null)
+            {
+                _csc = new CompilerSettings($"{customRoslynCompilerLocation}\\csc.exe", ttl);
```

Comment:
Msbuild 14 supports C# 6 and it was shipped 2 years ago, so it should be safe to assume that. And it's VS2015 solution, I believe msbuild was updated in update 2 or 3 to support C#6.